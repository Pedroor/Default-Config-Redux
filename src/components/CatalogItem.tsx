import React, { useEffect, useState, useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { View, Text, TouchableOpacity } from 'react-native';
import { addProductToCartRequest } from '../store/modules/cart/action';
import { IProduct } from '../store/modules/cart/types';
import { IState } from '../store';

// import { Container } from './styles';
interface CatalogItemProps {
  product: IProduct;
}

function CatalogItem() {
  // const dispatch = useDispatch();

  // const hasFailedStockCheck = useSelector<IState, boolean>(state => {
  //   return state.cart.failedStockCheck.includes(product.id);
  // });

  // const handleAddProductToCart = useCallback(
  //   (product: IProduct) => {
  //     dispatch(addProductToCartRequest(product));
  //   },
  //   [dispatch],
  // );

  return (
    <View style={{ marginTop: 15 }}>
      {/* <View style={{ flexDirection: 'row', justifyContent: 'center' }}>
        <Text style={{ paddingRight: 50 }}>{product.title}</Text>
        <Text>{product.price}</Text>
      </View> */}
      <View
        style={{
          justifyContent: 'center',
          alignItems: 'center',
          paddingTop: 20,
        }}>
        <TouchableOpacity
          style={{ width: '70%', height: 35, backgroundColor: 'green' }}
          // onPress={() => handleAddProductToCart(product)}
        >
          <Text style={{ textAlign: 'center' }}>Adicionar</Text>
        </TouchableOpacity>

        {/* {hasFailedStockCheck && (
          <Text style={{ color: 'red' }}>PRODUTO FORA DE ESTOQUE</Text>
        )} */}
      </View>
    </View>
  );
}

export default CatalogItem;
