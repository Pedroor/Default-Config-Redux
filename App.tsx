import React from 'react';
import { SafeAreaView, StyleSheet, Text } from 'react-native';
import { Provider } from 'react-redux';
import { Catalog } from './src/components/Catalog';
import { Cart } from './src/components/Cart';
import store from './src/store';

const App = () => {
  if (__DEV__) {
    require('react-devtools');
  }
  return (
    <Provider store={store}>
      <SafeAreaView style={styles.container}>
        <Text>OLA MUNDO</Text>
        <Catalog />
        <Cart />
      </SafeAreaView>
    </Provider>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default App;
