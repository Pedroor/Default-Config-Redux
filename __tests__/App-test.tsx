import React from 'react';

import { render } from '@testing-library/react-native';
import App from '../App';
import CatalogItem from '../src/components/CatalogItem';

describe('Render successfully', () => {
  it('renders App successfully', () => {
    const { toJSON } = render(<App />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders login button successfully', async () => {
    const { getByText } = render(
      <CatalogItem
        product={{
          id: 3,
          title: 'TÃªnis da moda',
          price: 500.0,
          hasProductOnStock: false,
        }}
      />,
    );

    const button = await getByText('Adicionar');

    expect(button).toBeTruthy();
  });
});
