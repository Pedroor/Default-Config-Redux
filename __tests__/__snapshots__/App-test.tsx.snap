// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render successfully renders App successfully 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    OLA MUNDO
  </Text>
  <View />
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingLeft": 40,
        }
      }
    >
      <Text
        style={
          Object {
            "paddingLeft": 15,
          }
        }
      >
        PRODUTO
      </Text>
      <Text
        style={
          Object {
            "paddingLeft": 15,
          }
        }
      >
        PREÃ‡O
      </Text>
      <Text
        style={
          Object {
            "paddingLeft": 15,
          }
        }
      >
        QUANTIDADE
      </Text>
      <Text
        style={
          Object {
            "paddingLeft": 15,
          }
        }
      >
        SUBTOTAL
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
